web_m2x_options_manager/models/m2x_create_edit_option.py

```diff
@@ -0,0 +1,168 @@
+from odoo.exceptions import ValidationError
+from odoo.tools.cache import ormcache
+from odoo.tools.safe_eval import safe_eval
+
+
+class M2xCreateEditOption(models.Model):
+    _name = "m2x.create.edit.option"
+    _description = 'Manage Options "Create/Edit" For Fields'
+
+    ir_model_field_id = fields.Many2one(
+        "ir.model.fields",
+        domain=[("ttype", "in", ("many2many", "many2one"))],
+        ondelete="cascade",
+        required=True,
+        string="Field",
+    )
+
+    ir_model_field_name = fields.Char(
+        related="ir_model_field_id.name",
+        store=True,
+        string="Field Name",
+    )
+
+    ir_model_id = fields.Many2one(
+        "ir.model",
+        ondelete="cascade",
+        required=True,
+        string="Model",
+    )
+
+    ir_model_name = fields.Char(
+        compute="_compute_ir_model_name",
+        inverse="_inverse_ir_model_name",
+        store=True,
+        string="Model Name",
+    )
+
+    option_create = fields.Selection(
+        [
+            ("none", "Do nothing"),
+            ("remove", "Remove"),
+            ("force_remove", "Force Remove"),
+        ],
+        default="remove",
+        help='Defines behaviour for "Create" option:\n'
+        "* Do nothing: nothing is done\n"
+        "* Remove: option 'Create' is set to False only if not already"
+        " present\n"
+        "* Force Remove: option 'Create' is always set to False",
+        required=True,
+        string="Create Option",
+    )
+
+    option_create_edit = fields.Selection(
+        [
+            ("none", "Do nothing"),
+            ("remove", "Remove"),
+            ("force_remove", "Force Remove"),
+        ],
+        default="remove",
+        help='Defines behaviour for "Create & Edit" option:\n'
+        "* Do nothing: nothing is done\n"
+        "* Remove: option 'Create & Edit' is set to False only if not already"
+        " present\n"
+        "* Force Remove: option 'Create & Edit' is always set to False",
+        required=True,
+        string="Create & Edit Option",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        self.clear_caches()
+        return super().create(vals_list)
+
+    def write(self, vals):
+        self.clear_caches()
+        return super().write(vals)
+
+    def unlink(self):
+        self.clear_caches()
+        return super().unlink()
+
+    @api.depends("ir_model_id")
+    def _compute_ir_model_name(self):
+        for opt in self:
+            opt.ir_model_name = opt.ir_model_id.model
+
+    def _inverse_ir_model_name(self):
+        ir_model_obj = self.env["ir.model"]
+        for opt in self:
+            # Using limit=1 since there's a SQL constrain to grant uniqueness
+            opt.ir_model_id = ir_model_obj.search(
+                [("model", "=", opt.ir_model_name)], limit=1
+            )
+
+    @api.constrains("ir_model_name", "ir_model_field_name")
+    def _check_model_and_field_names(self):
```

Comment:
can't you avoid this by throwing an error on the inverse when no record is found?

Reply:
@simahawk I believe a constrain makes it clearer and more manageable